code/mob/living/silicon/robot.dm

```diff
@@ -247,30 +246,26 @@
 		hud.update_pulling()
 
 	death(gibbed)
-		if (src.mainframe)
-			logTheThing("combat", src, null, "'s AI controlled cyborg body was destroyed [log_health(src)] at [log_loc(src)].") // Brought in line with carbon mobs (Convair880).
-			src.mainframe.return_to(src)
-		setdead(src)
-		borg_death_alert()
-		src.canmove = 0
-
-		if (src.camera)
-			src.camera.camera_status = 0.0
-
-		src.sight |= SEE_TURFS | SEE_MOBS | SEE_OBJS
-
-		src.see_in_dark = SEE_DARK_FULL
-		if (client?.adventure_view)
-			src.see_invisible = 21
-		else
-			src.see_invisible = 2
-
-		logTheThing("combat", src, null, "was destroyed [log_health(src)] at [log_loc(src)].") // Only called for instakill critters and the like, I believe (Convair880).
+		logTheThing("combat", src, null, "was destroyed at [log_loc(src)].")
+		src.mind?.register_death()
+		if (src.mind?.special_role)
+			src.handle_robot_antagonist_status("death", 1)
+		src.borg_death_alert()
+		if (!gibbed)
+			src.visible_message("<span class='alert'><b>[src]</b> falls apart into a pile of components!</span>")
+			var/turf/T = get_turf(src)
+			for(var/obj/item/parts/robot_parts/R in src.contents)
+				R.set_loc(T)
```

Comment:
This is probably safer than using the same loc as the mob (which would let you have them gib in lockers), as I can definitely see some issues with e.g. gib in a pod putting all of the random bits within the pod and unretrievable.

Might cause the odd weird interaction here or there, but overall I'd prefer that over the edge-case bugs, so this seems fine.

Reply:
Yeah, if the constant 'mechanics-deconstructing-shit-with-people-inside' issues have taught me anything, I'd rather err on the side of safety. I'd be open to using `loc` directly in the future, but it will require a lot of testing.